File: ServiGo/backend/apps/usuarios/repositories/notificacion_repository.py
from ..models import Notificacion


class NotificacionRepository:

    @staticmethod
    def crear(usuario, reserva, tipo, mensaje):
        return Notificacion.objects.create(
            usuario=usuario,
            reserva=reserva,
            tipo=tipo,
            mensaje=mensaje
        )

    @staticmethod
    def por_usuario(usuario_id):
        return Notificacion.objects.filter(
            usuario_id=usuario_id
        ).order_by("-creada_en")

File: backend/src/repositories/notificacion.repository.ts
import { Notificacion } from "../models/Notificacion";

export class NotificacionRepository {
  static crear(data: any) {
    return Notificacion.create(data);
  }
  static porUsuario(usuarioId: string) {
    return Notificacion.find({ usuario: usuarioId }).sort({ creadaEn: -1 }).lean();
  }
}
